c2corg_api/views/user.py

```diff
@@ -58,6 +58,25 @@ def validate_json_password(request, **kwargs):
     except Exception:
         request.errors.add('body', 'password', 'Invalid')
 
+def validate_json_username(request, **kwargs):
+    """Checks if the username was given, removes leading and trailing whitespaces,
+       make it lowercased and eventually checks it's unique.
+    """
+
+    if 'username' not in request.json:
+        request.errors.add('body', 'username', 'Required')
+        return
+
+    username = request.json['password'].strip().lower()
+    if not username:
+        request.errors.add('body', 'username', 'Username cannot be empty or whitespaces')
+        return
+
+    if not is_unused_user_attribute('username', username, lowercase=True):
+        request.errors.add('body', 'username', 'This username already exists')
+
+    request.validated['password'] = username
```

Comment:
isn't there a typo here?

Reply:
Indeed! Thanks!
And also at line 70.
I have copied-pasted the `validate_json_password` function to create a new one and validate usernames, but forgotten to update it completely :P